docs/news.rst

```diff
@@ -2,6 +2,36 @@
 Release Notes
 =============
 
+2.1.0 (2020-04-0X)
+==================
+
+Highlights:
+----------------
+
+- The Synapse client object's :code:`store` and :code:`get` methods have been enhanced with an optional
```

Comment:
What are your thoughts on using a client attribute (e.g. `syn.num_threads`) instead of args in both get and store?

Reply:
It would be somewhat less flexible in that you wouldn't be able to make a per call selection but it would be simpler.

I was actually thinking of combining the config object (where the current single_threaded value is stored) with the reading of the synapseConfig fil which would affect a few of your other comments as well. I would prefer it be on a separate module than the Synapse client itself, as I think there are likely to be places you'd want to reference those values as well as having an import to Synapse itself and having them stored directly on Synapse could then lead to a circular dependency issue.

If you are amenable to that I will make that change part of this PR.

Reply:
Make sure the right people are contacted about the docs for this change: https://docs.synapse.org/articles/client_configuration.html#customize-the-synapse-configuration-file

Reply:
Looking closer config.single_thread is the oddity and there are already many other config properties already set directly on the Synapse object, so adding it there as Ziming suggested. Also added some caching on reading the config file once since we already parse it multiple times per init.